test/org/sagebionetworks/bridge/services/ParticipantServiceTest.java

```diff
@@ -896,6 +814,268 @@ public void sendNotification() {
+        setupExternalIdTest(false, true);
+        when(lookup.getString(EXTERNAL_IDENTIFIER)).thenReturn(EXTERNAL_ID);
+        when(optionsService.getOptions(HEALTH_CODE)).thenReturn(lookup);
+        
+        participantService.updateParticipant(STUDY, CALLER_ROLES, NO_ID_PARTICIPANT);
+        
+        verifyNotSetAsReservation();
+        verifySetAsOption(null);
+    }
+
+    @Test(expected = BadRequestException.class)
+    public void updateExternalIdValidatedNotRequiredNoValue() {
+        setupExternalIdTest(true, false);
+        when(lookup.getString(EXTERNAL_IDENTIFIER)).thenReturn(EXTERNAL_ID);
+        when(optionsService.getOptions(HEALTH_CODE)).thenReturn(lookup);
+        
+        participantService.updateParticipant(STUDY, CALLER_ROLES, NO_ID_PARTICIPANT);
+    }
+    
+    @Test
+    public void updateExternalIdNotValidatedNotRequiredNoValue() {
+        setupExternalIdTest(false, false);
+        when(lookup.getString(EXTERNAL_IDENTIFIER)).thenReturn(EXTERNAL_ID);
+        when(optionsService.getOptions(HEALTH_CODE)).thenReturn(lookup);
+        
+        participantService.updateParticipant(STUDY, CALLER_ROLES, NO_ID_PARTICIPANT);
+        
+        verifyNotSetAsReservation();
+        verifySetAsOption(null);
+    }
+    
+    // Updating the participant with an external ID when one was not provided on creation
+    // (even if that would have been prevented by the code, just in case)
+    
+    // It's after creation so we just don't do it... of all the tests, this one is 
+    // debatable. If the value is supplied and its correct, even though it was required
+    // at sign up and should be there, we should probably add it now.
+    @Test
+    public void updateExternalIdValidatedRequiredNewValue() {
+        setupExternalIdTest(true, true);
+        
+        participantService.updateParticipant(STUDY, CALLER_ROLES, NEW_ID_PARTICIPANT);
+        
+        verifyNotSetAsReservation();
+        verifyNotSetAsOption();
+    }
+
+    @Test
+    public void updateExternalIdNotValidatedRequiredNewValue() {
+        setupExternalIdTest(false, true);
+        
+        participantService.updateParticipant(STUDY, CALLER_ROLES, NEW_ID_PARTICIPANT);
+        
+        verifyNotSetAsReservation();
+        verifySetAsOption("newExternalId");
+    }
+
+    @Test
+    public void updateExternalIdValidatedNotRequiredNewValue() {
+        setupExternalIdTest(true, false);
+        
+        participantService.updateParticipant(STUDY, CALLER_ROLES, NEW_ID_PARTICIPANT);
+        
+        verifyIdAssignedWithoutReservation("newExternalId");
+        verifyNotSetAsOption();
+    }
+    
+    @Test
+    public void updateExternalIdNotValidatedNotRequiredNewValue() {
+        setupExternalIdTest(false, false);
+        
+        participantService.updateParticipant(STUDY, CALLER_ROLES, NEW_ID_PARTICIPANT);
+        
+        verifyNotSetAsReservation();
+        verifySetAsOption("newExternalId");
+    }    
```

Comment:
The introduction of these behavior flags in particular has caused exponential growth in the tests, but I will admit I don't have an answer about how to reduce the complexity right now (other than to split the create/update code paths, which would make it easier to reason about but wouldn't reduce the number of tests).